app/src/store/middleware/log.ts

```diff
@@ -0,0 +1,54 @@
+import { type Middleware } from '@reduxjs/toolkit';
+import { type Dispatch, type State } from '../app';
+
+export type LogOptions = Readonly<{
+  logStackTrace?: boolean;
+}>;
+
+export const logMiddlewareDefaultOptions: LogOptions = Object.freeze({
+  logStackTrace: false,
+});
+
+export const createLogMiddleware: (
+  options?: LogOptions,
+) => Middleware<{}, State, Dispatch> =
+  ({ logStackTrace } = logMiddlewareDefaultOptions) =>
+  () =>
+  (next) =>
+  (action) => {
+    let message: string = action.type;
+
+    if (action.type.match(/\/rejected$/g)) {
+      const error = action.payload || action.error;
+
+      const filteredError = logStackTrace
+        ? error
+        : filterOutKey('stack', error);
+
+      message += ' ' + JSON.stringify(filteredError, null, 2);
+    }
+
+    console.log(message);
```

Comment:
What does this look like from the console? Would you be able to see where the log is coming from, or would this only show that all logs are coming from `log.ts`?

Reply:
Weird, no line numbers or source file info?

Reply:
Not in the terminal, no.